nixos/modules/services/networking/mosh.nix

```diff
@@ -0,0 +1,19 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg  = config.services.mosh;
```

Comment:
I think I agree with @fpletz about using `programs`; calling mosh a service is a bit misleading, and anybody who goes looking for mosh in the options list will probably just search for it by name.